source/MissionAction.cpp

```diff
@@ -74,6 +128,12 @@ void MissionAction::Load(const DataNode &node)
 
 	for(const DataNode &child : node)
 		LoadSingle(child);
+
+	// Collapse pure-text dialog (no conditions or phrases). This is necessary to handle saved missions.
+	// It is also an optimization for the most common case in game data files.
+	dialogText = CollapseDialog(nullptr, nullptr);
+	if(!dialogText.empty())
+		dialog.clear();
```

Comment:
```suggestion
	dialog.clear();
```
Unnecessary check

Reply:
Incorrect. The dialogText is only empty if the dialog was entirely plain text (no phrases or conditions).

Reply:
ah yes, I read it wrong